specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2020-06-01-preview/rbac.json

```diff
@@ -0,0 +1,552 @@
+        },
+        "principalId": {
+          "type": "string",
+          "description": "The unique identifier for the associated AAD principal in the AAD graph to which access is being granted through this Role Assignment. Tenant ID for the principal is inferred using the tenant associated with the subscription."
+        }
+      },
+      "x-ms-azure-resource": true,
+      "description": "Role Assignment Properties."
+    },
+    "SqlRoleAssignmentResponse": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The fully qualified GUID of the Role Assignment containing the resource path and the ID supplied for this Role Assignment during create."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The ID supplied for this Role Assignment during create."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Microsoft.DocumentDB/databaseAccounts/roleAssignments."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SqlRoleAssignmentProperties",
+          "description": "Properties related to Role Assignment."
+        }
+      },
+      "x-ms-azure-resource": true,
+      "description": "An Azure Cosmos DB Role Assignment"
+    },
+    "SqlRoleAssignmentListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SqlRoleAssignmentResponse"
+          }
+        }
+      },
+      "description": "The relevant Role Assignments."
+    }
+  },
+  "parameters": {
+    "roleDefinitionIdParameter": {
+      "name": "roleDefinitionId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "description": "The GUID for the Role Definition."
+    },
+    "databaseAccountNameParameter": {
+      "name": "databaseAccountName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "description": "Name of Cosmos DB database account."
+    },
+    "roleAssignmentIdParameter": {
+      "name": "roleAssignmentId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "description": "The GUID for the Role Assignment."
+    },
+    "filterRoleDefinitionParameter": {
+      "name": "$filter",
+      "in": "query",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "description": "The filter to be applied on the request. Use $filter=type eq 'BuiltInRole|CustomRole' to filter by type. Use $filter=roleName eq {name} to retrieve Role Definition with the given name."
+    },
+    "filterRoleAssignmentParameter": {
+      "name": "$filter",
```

Comment:
@MehaKaushik we haven't implemented filter yet in RP. Will this be a problem?

Reply:
Lets not add anything which is not in Prod yet. Let us revert this.